ps_socialfollow.php

```diff
@@ -93,14 +96,16 @@ public function uninstall()
 
     public function getContent()
     {
+        $this->validation_message = '';
         if (Tools::isSubmit('submitModule')) {
-            $this->updateFields();
-            $this->_clearCache('*');
+            if ($this->updateFields() === true) {
+                $this->_clearCache('*');
 
-            Tools::redirectAdmin($this->context->link->getAdminLink('AdminModules').'&configure='.$this->name.'&tab_module='.$this->tab.'&conf=4&module_name='.$this->name);
+                Tools::redirectAdmin($this->context->link->getAdminLink('AdminModules') . '&configure=' . $this->name . '&tab_module=' . $this->tab . '&conf=4&module_name=' . $this->name);
```

Comment:
Nice-to-have: Could we use sprintf and stop relying on string concatenations?

Reply:
See slack...

Reply:
As you mentioned in Slack,
```
sprintf(
    '%1$s&configure=%2$s&tab_module=%3$s&conf=4&module_name=%2$s',
    $this->context->link->getAdminLink('AdminModules'),
    $this->name,
    $this->tab
);
```
is not very good to read either 😅 

What we are looking for is a way to make it easy for reader to understand "what are the parameters used" and "what values are they"

Maybe something like
```
$url = $this->context->link->getAdminLink('AdminModules') .
'&configure=' . $this->name .
'&tab_module=' . $this->tab .
'&conf=4&module_name=' . $this->name;

Tools::redirectAdmin($url);
```
would be better? one line per parameter => it allows reader to check easily each part of the URL

Or we could also use `http_build_query`
Or we could also use built-in `$params` argument:
https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Link.php#L724
```
public function getAdminLink($controller, $withToken = true, $sfRouteParams = [], $params = [])
```

🔽 
```
$url = $this->context->link->getAdminLink('AdminModules');
$parameters = ['configure' => $this->name,
'tab_module' => $this->tab,
'conf' => 4,
'module_name' => $this->name];

Tools::redirectAdmin($url, true, [], $parameters);
```

Reply:
👍 Looks like the good idea.